src-web/components/kappnav/modals/NavModal.js

```diff
@@ -148,29 +149,50 @@ class NavModal extends React.PureComponent {
             open={open} 
             onClose={() => this.toggleModal(false)}>
 
-            <ModalHeader buttonOnClick={this.handleOpen.bind(this, false)} iconDescription={this.props.closeButtonLabel}>
-              <div>
-                <p className='bx--modal-header__label'>
-                  {this.props.modalLabel}
-                </p>
-                <p className='bx--modal-header__heading'>
-                  {this.props.modalHeading}
-                </p>
+          <ModalHeader buttonOnClick={this.handleOpen.bind(this, false)} iconDescription={this.props.closeButtonLabel}>
+            <div>
+              <p className='bx--modal-header__label'>
+                {this.props.modalLabel}
+              </p>
+              <p className='bx--modal-header__heading'>
+                {this.props.modalHeading}
+              </p>
+              {/* adding the subheader to the modal */}
+              <div className="sub-header">
+                <div>
+                  {msgs.get('formfield.header.' + this.props.kind) }
+                </div>
+                <div contentEditable = {false} data-text="enter name" id= "modal-name">
+                </div>
               </div>
-              {
-                !hideJsonEditor && <div className='toggle'>
-                    <p>
-                      {msgs.get('mode.json')}
-                    </p>
-                    <Toggle id='json-toggle' onToggle={this.handleToggle} labelA={msgs.get('off')} labelB={msgs.get('on')}/>
+            </div>
+            {
+              !hideJsonEditor && <div className='toggle'>
+                <p style={{ display: "inline" }}>
+                  {msgs.get('mode.json')}
+                </p>
+                {/* adding tooltip */}
+                <Tooltip
+                  iconDescription={msgs.get('formtip.tooltip')}
+                  triggerText=''
+                  direction='right'
+                  showIcon={true}>
+                  <p dangerouslySetInnerHTML={{ __html: msgs.get('tooltip.jsonmode') }} />
+                  <div className="bx--tooltip__footer">
+                    <a className="bx--link" target="_blank" href={this.props.learnmoreURL}>
```

Comment:
We need to add rel=”noopener noreferrer” to the tag because without it we have a security vulnerability.